From bdac3dcb2f18fbdc24fcd380591474bb539f006f Mon Sep 17 00:00:00 2001
From: reaper <theunbounddeveloper@outlook.com>
Date: Sun, 23 Aug 2026 21:01:53 +0200
Subject: [PATCH] Update amtm 7.0

Updated amtm firmware file to revision 9.

(cherry picked from commit c0763a3475491fdc4e7eeb234094f45341450191)
---
 release/src/router/others/amtm | 29 +++++++++++++++--------------
 1 file changed, 15 insertions(+), 14 deletions(-)

diff --git a/release/src/router/others/amtm b/release/src/router/others/amtm
index d461f24f77..2fe139e66b 100755
--- a/release/src/router/others/amtm
+++ b/release/src/router/others/amtm
@@ -10,7 +10,7 @@
 
 add=/jffs/addons/amtm
 amtmURL=https://fwupdate.asuswrt-merlin.net/amtm_fw
-amtmRev=8
+amtmRev=9
 
 a_m(){ [ -z "$am" ] && am=$1 || am="$am\\n$1";}
 c_nl(){ [ -n "$(tail -c2 "$1")" ] && echo >> "$1";}
@@ -38,7 +38,13 @@ c_t(){
 	if [ -f "${add}"/.amtm_theme ]; then
 		. "${add}"/.amtm_theme
 	else
-		[ -s "$divconf" ] && theme="$(grep "THEME=" "$divconf" | sed -e 's/THEME=//')"
+		if [ -s "$divconf" ]; then
+			while read -r line; do
+				case "$line" in
+					THEME=*) theme="${line#THEME=}"; break ;;
+				esac
+			done < "$divconf"
+		fi
 	fi
 	[ "$theme" ] && theme_$theme || theme_amtm new
 }
@@ -195,19 +201,14 @@ init_amtm(){
 run_amtm(){
 	if [ ! -f "${add}"/a_fw/amtm.mod ] || [ -f /jffs/scripts/amtm ]; then
 		init_amtm
-	elif [ -z "$1" ]; then
-		[ "$am" ] && show_amtm "$am" || show_amtm menu
-	elif [ "$1" = tpu ]; then
-		su=1;suUpd=0;updErr=;tpu=1
-		> /tmp/amtm-tpu-check
-		show_amtm >/dev/null 2>&1
-	elif [ "$1" = updcheck ]; then
-		su=1;suUpd=0;updErr=;tpu=1;updcheck=1
-		echo "Available script updates:" >/tmp/amtm-tpu-check
-		update_amtm
-		show_amtm
 	else
-		show_amtm "$1"
+		case "$1" in
+			"") 		[ "$am" ] && show_amtm "$am" || show_amtm menu;;
+			tpu) 		tpu_check;;
+			updcheck) 	upd_check;;
+			autoupdate)	auto_script_update;;
+			*)			show_amtm "$1";;
+		esac
 	fi
 }
 [ -f "${add}"/a_fw/amtm.mod ] && . "${add}"/a_fw/amtm.mod
-- 
2.25.1

